ui/src/app/applications/components/application-resource-tree/application-resource-tree.tsx

```diff
@@ -170,6 +257,38 @@ function renderFilteredNode(node: {count: number} & dagre.Node, onClearFilter: (
     );
 }
 
+function renderGroupedNodes(props: ApplicationResourceTreeProps, node: {count: number} & dagre.Node & ResourceTreeNode) {
+    const indicators = new Array<number>();
+    const groupdedNodeIds = node.groupdedNodeIds;
+    let count = Math.min(node.count - 1, 3);
+    while (count > 0) {
+        indicators.push(count--);
+    }
+    return (
+        <React.Fragment>
+            <div className='application-resource-tree__node' style={{left: node.x, top: node.y, width: node.width, height: node.height}}>
+                <div className='application-resource-tree__node-kind-icon'>
+                    <ResourceIcon kind={node.kind} />
+                    <br />
+                    <div className='application-resource-tree__node-kind'>{ResourceLabel({kind: node.kind})}</div>
+                </div>
+                <div className='application-resource-tree__node-content-wrap-overflow'>
+                    <a className='application-resource-tree__node-title' onClick={() => props.onGroupdNodeClick && props.onGroupdNodeClick(groupdedNodeIds)}>
+                        click to show details of {node.count} collapsed {node.kind}
+                    </a>
+                </div>
+            </div>
+            {indicators.map(i => (
+                <div
```

Comment:
If 5 resources are grouped, and say, 3 are healthy and 2 are not, and I apply a filter to show only healthy resources, then should the icon change to show the filter icon within this group container?

Reply:
As discussed in huddle, this works as expected.